hd/address.go

```diff
@@ -282,9 +282,20 @@ func CalcSha512(buf []byte) []byte {
 }
 
 func ReverseBytes(buf []byte) []byte {
-	res := []byte{}
-	for i := len(buf) - 1; i >= 0; i-- {
-		res = append(res, buf[i])
+	var res []byte
+	if len(buf) == 0 {
+		return res
+	}
+
+	// Walk till mid-way, swapping bytes from each end:
+	// b[i] and b[len-i-1]
```

Comment:
Yes, nice swap algo to avoid multiple allocations.

Seems adapted from an in-place swap, you could probably do it simpler with one assignment per step in the for loop at the same efficiency.